### ISSUE TITLE
Gotcha: No error message when requesting different fields on related query and mutation

### ISSUE THREAD

**Issue by groeneman** [apollographql/apollo-client#973]
When using `updateQueries` to update `observableQuery` state after a mutation, `react-apollo` can incorrectly suggest an `observableQuery` is `loading` when the mutation fails to request some fields requested by the query. Imagine the following query and mutation:

```
query LoadAnimals {
  animals { id name color }
}
```

```
mutation AddAnimal($animal AnimalInput!) {
  addAnimal(input: { animal: $animal }) {
    animal { id name } # note absence of the 'color' attribute here!
  }
}
```

Further assume that the `AddAnimal` mutation uses `updateQueries` to update the `LoadAnimals` query, appending the result of `AddAnimal` to the collection using something like:

```
updateQueries: {
  LoadAnimals: (prev, { mutationResult }) => [...prev, mutationResult.data.addAnimal.animal]
}
```

The next time [QueryManager.getCurrentQueryResult#728](https://github.com/apollostack/apollo-client/blob/master/src/core/QueryManager.ts#L728) is called for the `LoadAnimals` query, `readQueryFromStore` will raise an exception because the `color` field can't be found on the newly added animal, and therefore the return value of `getCurrentQueryResult` will be `{ data: {}, partial: true }`. Downstream, this will cause the loading flag to be set on [ObservableQuery.currentResult#136](https://github.com/apollostack/apollo-client/blob/1196777131838e90530e00820d7060360298f7fc/src/core/ObservableQuery.ts#L136), even though nothing is loading. In many cases, the result is an inexplicable loading spinner (for something that will in fact never load) instead of a helpful error message about the missing field on the mutation. I was at a complete loss for how to resolve this in my own app until I dove into the `apollo-client` source.

A helpful warning about the missing field would be good here.

**Comment by helfer** [comment]
@mgroeneman good catch! I thought we had some logic in place to avoid that exact problem, but maybe it doesn't inspect arrays. I think it should definitely be fixed. Would you be interested in working on a fix with us? 🙂

**Comment by mgroeneman** [comment]
@helfer , I'd be happy to work on a PR. If you've got any advice on where in the code to insert a check for this issue, that'd be a huge help!

**Comment by cesarsolorzano** [comment]
I think this should be closed in favor of #1273
